### DISCUSSION TITLE
== Error ==  system call for: "" finished abnormally, OS return value: 21

### DISCUSSION THREAD

**Discussion by AlexandrosMosca** [ablab/spades#1096] [Q&A]
### Description of bug

Hello,
Sorry in advance if this question could be not so relevant. I ran the spades commands on my fq.gz data that were filtered from host contamination sequences and that I obtained after having been used bwa and samtools. Although the message error is clear, I was wondering if there's still a chance to perform the assembly analysis with spades in these data with a specific argument in the commands. 

### spades.log

[spades.log](https://github.com/ablab/spades/files/10884496/spades.log)

### params.txt

[params.txt](https://github.com/ablab/spades/files/10884516/params.txt)

### SPAdes version

3.15.5

### Operating System

3.11.0

### Python Version

Linux-4.18.0-425.10.1.el8_7.x86_64-x86_64-with-glibc2.28

### Method of SPAdes installation

conda

### No errors reported in spades.log

- [x] Yes

**Comment by asl** [comment]
The log reads:

```
  4:17:30.729   125G / 181G  ERROR   General                 (hammer_tools.cpp          : 191)   Pair of read files /mnt/shared/scratch/amosca/transfer_test/C4-A2-S32_1.filtered_2.fq.gz and /mnt/shared/scratch/amosca/transfer_test/C4-A2-S32_1.filtered_1.fq.gz contain unequal amount of reads
```

So your input data is corrupted, your left reads should correspond to the right ones. Ensure that any filtering you're doing is paired-end aware. Also there are some tools around that might "fix the pairedness" relying on the read names. You might give them a try.

**Reply by AlexandrosMosca** [reply]
Thank you very much. I've just checked my filtering commands and it looks that they are paired-end aware. I've just seen that the filtered fastq files have this title (@):

```
 @VH00754:1:AAAT5JJHV:2:2608:64377:36837/1
GTGATTGAGCACCATGTGCAGGCATCATGATCGAAAATCACCCCAGTCCTGCAGCCTGCAATTTGCGCTGCAATCCCATGGGCAACACCGATGCACCCTAAACCCTAAACCCTAACCTGAAACCCTAAGCAGTTTCTATTTAAAATGATG
+
CC;CC-CCCCC-CCCC;CCC-CCCCCCC-C;CCCCC;CCCCC-CC;;CCCCCCCCCCC;CC;CCC;CCCCC;CCCCCCCCCC-;CC-CCCC;CC;CCC;;CCCCCCCCCC;;CCC;CC;CC;CCCCCCC-CCC;CCC-CCCCCCCC-CC;

```

Where it seems that /1 are the forward reads and /2 the reverse ones.


```
@VH00754:1:AAAT5JJHV:1:2607:35008:11147/2
CGCGCAGTCCTCGGCACGATCGCGGGCATGATGAGCAAGTCGGACACGCTGGGGTATCTCGGTTCGTACAAGGTGCCGGAAGTCGTGCTCGGCGTGACCTCGTTCCTGCTGGCTCCTAAACCCTAACCCTAAACCCTAAACCCTAACCC
+
CCCCCCCCCCCCCC;CCCCCCCCCCCCCCCCCCCCCCCCCCCCCC;CCCCCCCCCCCCCCCCC-;C;;CCC-C;CC;C-CCC;CCCC-CCCCCCCC--CCCCCCCCCCCCC;CCCCCC--CCCCC;CCCCCCCCCCCC;-CCCC-C-CC
```


I double-checked in the input files that I used for the filtering analysis and it seems all right:

```
@VH00754:1:AAAT5JJHV:1:1101:23420:1000 **_1_**:N:0:AGTACAGTTC+CCTTGGCTTC
GCCGTCGAGAACATCATCGACTCGGTGGCGCGCGAGCTGAAGCGCGATCCGCTCGACGTGCGGCGCGTCAACTTCTACGGCAAGGCGGAGAACAACGTCACGCCCTACCGGCAGACGGTGACCGACAACATCGTGCACGAGCTGGTGGCCG
+
!CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC-CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC

@VH00754:1:AAAT5JJHV:1:1101:18459:1000 **_2_**:N:0:AAGAGATCAC+GTGTGATAAC
TGCAGCTCCGCCAGCAGCTTGTCATTGTGCTTGGTCAGCTGCCGCAGGTTGGAACCCTCGAGCGCGTAGACCTCGTTCATCCGCGTGGCCTCGGCGGCAAGCACCGCGAAGTTGCCGTCCTTGCCGGGACTCGGCGAGGAAATGACGCGAT
+
CCCCCCCCCCCCCCCCC-;CCCCCCCCCCC;CCCCCCC;;CCCCCCCCC-CCCCCCCC;CCCCCCCCCCCCCCCCCCCCCCCCCCCCC;CCCCCCCCCCC;CCCCCCCC-;CCCCCCCCCCCCCCCCC-CCCCCCCCCCCCCC;CCCCCCC
```


Could fastp assess and remove this potential error? Can you suggest any other tools? I'm very new to this kind of analysis

**Reply by asl** [reply]
Well, `@VH00754:1:AAAT5JJHV:2:2608:64377:36837/1` and `@VH00754:1:AAAT5JJHV:1:2607:35008:11147/2` are different read names. The parts before `/1` and `/2` should certainly match. You need to ensure your filtering is paired-end aware. Maybe your input files prior to filtering are already corrupted.
